### PR TITLE
Pseudo code is code component

### LINKED ISSUE #4177: Recovery-31: Use of <CODE BEGINS> and <CODE ENDS> tags
The recovery draft have significant amount of pseudo code. Per the IETF trust legal provision (TLP) (https://trustee.ietf.org/license-info/IETF-TLP-5.pdf) code components have a specific license. The TLP is also clear that anything enclosed in `<CODE BEGINS>` and `<CODE ENDS>` is to be considered code components. Thus for clarity of identification I would propose that the psudocode in this document uses these tags.

### PR DESCRIPTION
**Pull request by janaiyengar**
Closes #4177.

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -1150,6 +1150,9 @@ This document has no IANA actions.
 We now describe an example implementation of the loss detection mechanisms
 described in {{loss-detection}}.
 
+The pseudocode snippets in this section are licensed as Code Components; see the
+copyright notice above.
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md:1154]:
```suggestion
copyright notice.
```

I get that we can't cite this directly, but I think we're OK not saying "above".

File: draft-ietf-quic-recovery.md [hunk 2]
```diff
@@ -1605,6 +1608,9 @@ OnPacketNumberSpaceDiscarded(pn_space):
 We now describe an example implementation of the congestion controller described
 in {{congestion-control}}.
 
+The pseudocode snippets in this section are licensed as Code Components; see the
+copyright notice above.
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md:1612]:
```suggestion
copyright notice.
```

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -1150,6 +1150,9 @@ This document has no IANA actions.
 We now describe an example implementation of the loss detection mechanisms
 described in {{loss-detection}}.
 
+The pseudocode snippets in this section are licensed as Code Components; see the
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md:1153]:
```suggestion
The pseudocode in this section is licensed as Code Components; see the
```

File: draft-ietf-quic-recovery.md [hunk 4]
```diff
@@ -1605,6 +1608,9 @@ OnPacketNumberSpaceDiscarded(pn_space):
 We now describe an example implementation of the congestion controller described
 in {{congestion-control}}.
 
+The pseudocode snippets in this section are licensed as Code Components; see the
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md:1611]:
```suggestion
The pseudocode in this section is licensed as Code Components; see the
```

**Review by martinthomson — APPROVED** [review]
This works for me.  Thanks for writing it up.

Some word deletion suggestions only.

File: draft-ietf-quic-recovery.md [hunk 5]
```diff
@@ -1150,6 +1150,9 @@ This document has no IANA actions.
 We now describe an example implementation of the loss detection mechanisms
 described in {{loss-detection}}.
 
+The pseudocode in this section is licensed as Code Components; see the
```
Comment (MikeBishop) [inline draft-ietf-quic-recovery.md:1153]: The agreement in number between "The pseudocode" and "Components" is jarring.  I'd go either....
```suggestion
The pseudocode in this section is licensed as a Code Component; see the
```
or
```suggestion
The sections which follow contain pseudocode.  The contents of these sections are
licensed as Code Components; see the
```
Given that it's multiple blocks, I think keeping it plural (as it originally was) might be better, but I agree on finding a word besides "snippets."

**Review by MikeBishop — APPROVED** [review]

File: draft-ietf-quic-recovery.md (hunk 5 above)
Comment (janaiyengar) [reply draft-ietf-quic-recovery.md]: I'm using "segments", see if that works.

**Comment by martinthomson** [comment]
I think that this is OK, but I want to point out that the pseudocode is pretty useless without the bits around it that explain when it is used and how.  Why not just label the entire section as a code component?

**Review by ianswett — APPROVED** [review]

**Comment by janaiyengar** [comment]
@martinthomson : The point of the issue was to protect the pseudo code under the license. The explanatory text doesn't need to be licensed any differently than the text in the rest of the document, IMO. I'm happy to take an editorial PR to change this as you would like though, if you think it's useful enough.

**Pull request merged**